src/components/Header/Header.tsx

```diff
@@ -0,0 +1,82 @@
+import { useContext, useEffect, useState } from 'react';
+import { TodosContext } from '../../context/TodosContext';
+import classNames from 'classnames';
+import { Actions } from '../../constants/Actions';
+
+const Header = () => {
+  const context = useContext(TodosContext);
+
+  if (!context) {
+    throw new Error('TodosContext must be used within a TodosProvider');
+  }
```

Comment:
You do the same in every component

To follow DRY principal create a custom hook `useTodosContext`:

```
function useTodosContext() {
  const context = useContext(TodosContext);

  if (!context) {
    throw new Error('TodosContext must be used within a TodosProvider');
  }
  
  return context;
}
```
And in component `const context = useTodosContext();` it always will be defined